src/toolkit/metadata.cc

```diff
@@ -0,0 +1,174 @@
+#include <math.h>
+#include <stdio.h>
+#include <sstream>
+
+#include "agent.h"
+#include "context.h"
+#include "metadata.h"
+namespace cyclus {
+namespace toolkit {
+
+enum type_hash {
+  s,  // string
+  b,  // boolean
+  i,  // int
+  u,  // uint
+  d   // double
+};
+
+std::unordered_map<std::string, int> type_map = {
+    std::make_pair("s", s),  // string
+    std::make_pair("b", b),  // boolean
+    std::make_pair("i", i),  // int
+    std::make_pair("u", u),  // uint
+    std::make_pair("d", d)   // double
+};
+
+const std::list<std::string> usages = std::list<std::string>(
+    {"decommission", "deployment", "timestep", "throughput"});
+
+Metadata::Metadata() {}
+Metadata::~Metadata() {}
+
+void Metadata::RecordMetadata(Agent* agent) {
```

Comment:
I'm not sure this is the best function name.  Maybe `RegisterMetadata`?  No recording of metadata happens here, this function simply adds it to the output tables to be recorded in the future.

Reply:
I suppose you are basing it on the naming of `Datum::Record()` which is perhaps also poorly named...  Although in that case one can interpret it as "start recording"